server/controllers/forgotPassword.js

```diff
@@ -0,0 +1,66 @@
+import { prisma } from "../app.js";
+import bcrypt from "bcryptjs";
+import { v4 as uuidv4 } from "uuid";
+import { sendResetEmail } from "../utils/email.js";
+
+export const requestPasswordReset = async (req, res) => {
+    const { email } = req.body;
+    const user = await prisma.mainuser.findUnique({ where: { email } });
+    if (!user) {
+        return res.status(404).json({ message: "User not found" });
+    }
+
+    const token = uuidv4();
+    const hashedToken = await bcrypt.hash(token, 10);
+
+
+    // Ensure any previous tokens for this user are deleted
+    await prisma.passwordResetToken.deleteMany({
+        where: { userId: user.email }
+    });
+
+   const passwd = await prisma.passwordResetToken.create({
+        data: {
+            token: hashedToken,
+            userId: user.email,
+            expiresAt: new Date(Date.now() + 3600000), // 1 hour from now
+        }
+    });
+
+    const resetUrl = `http://192.168.93.236:5173/reset-password/${token}/${passwd.id}`;
```

Comment:
Is this URL correct? If so, are we redirecting to a different server?

Reply:
No  It wouldn't,    Actuallly react provide us two URL for rendering our site - one is localhost and another for over network.     
![image](https://github.com/user-attachments/assets/7061c122-8bb9-4d34-a007-1e11f517bc34)

I used  'Network URL '   here because generally multiple gmail accunts signed in the mobile devices, so  the reset link which we are sending on user mail will render the reset password page on mobile devices also.

Reply:
I made the changes.

Reply:
@dvjsharma , merge this PR.